wgengine/magicsock/magicsock.go

```diff
@@ -2099,6 +2114,29 @@ func (c *Conn) handleDiscoMessage(msg []byte, src netip.AddrPort, derpNodeSrc ke
 		return
 	}
 
+	// Emit information about the disco frame into the pcap stream
+	// if a capture hook is installed.
+	if cb := c.captureHook.Load(); cb != nil {
+		var (
+			l    bytes.Buffer
```

Comment:
call Grow on it first, before all the little writes?

but this is garbage per disco message. probably okay. Maybe just add a comment like "// Warning: alloc garbage. Acceptable when capturing."